pythnet/pythnet_sdk/src/wire.rs

```diff
@@ -0,0 +1,306 @@
+//! Pyth Wire Format
+//!
+//! Pyth uses a custom wire format when moving data between programs and chains. This module
+//! provides the serialization and deserialization logic for this format as well as definitions of
+//! data structures used in the PythNet ecosystem.
+//!
+//! See the `ser` submodule for a description of the Pyth Wire format.
+
+use {
+    borsh::BorshSerialize,
+    serde::Serialize,
+    wormhole_sdk::Vaa,
+};
+
+pub mod array;
+mod de;
+mod prefixed_vec;
+mod ser;
+
+pub use {
+    de::{
+        from_slice,
+        Deserializer,
+        DeserializerError,
+    },
+    prefixed_vec::PrefixedVec,
+    ser::{
+        to_vec,
+        to_writer,
+        Serializer,
+        SerializerError,
+    },
+};
+
+// Transfer Format.
+// --------------------------------------------------------------------------------
+// This definition is what will be sent over the wire (I.E, pulled from PythNet and
+// submitted to target chains).
+#[derive(BorshSerialize, Serialize)]
+pub struct AccumulatorProof<'a> {
+    magic:         [u8; 4],
+    major_version: u8,
+    minor_version: u8,
+    trailing:      &'a [u8],
+    proof:         v1::Proof<'a>,
+}
+
+// Proof Format (V1)
+// --------------------------------------------------------------------------------
+// The definitions within each module can be updated with append-only data without
+// requiring a new module to be defined. So for example, new accounts can be added
+// to the end of `AccumulatorAccount` without moving to a `v1`.
```

Comment:
Note that this is not correct, but Ali is modifying these to fit Hermes which is the right place for those changes anyway, so ignore these.